tests/utils/utils.go

```diff
@@ -542,16 +543,18 @@ func waitForTransaction(
 // It stops waiting when the context is canceled.
 // Takes a tx hash instead of the full tx in the subnet-evm version of this function.
 // Copied and modified from https://github.com/ava-labs/subnet-evm/blob/v0.6.0-fuji/accounts/abi/bind/util.go#L42
-func WaitMined(ctx context.Context, b bind.DeployBackend, txHash common.Hash) (*types.Receipt, error) {
+func WaitMined(ctx context.Context, rpcClient ethclient.Client, txHash common.Hash) (*types.Receipt, error) {
 	queryTicker := time.NewTicker(200 * time.Millisecond)
 	defer queryTicker.Stop()
 	cctx, cancel := context.WithTimeout(ctx, 10*time.Second)
 	defer cancel()
 
+	var receipt *types.Receipt
 	for {
-		receipt, err := b.TransactionReceipt(cctx, txHash)
+		var err error
+		receipt, err = rpcClient.TransactionReceipt(cctx, txHash)
 		if err == nil {
-			return receipt, nil
+			break
 		}
```

Comment:
We can probably combine the two loops by nesting the block height check here. One possible implementation would be to set a `receivedReceipt` bool and check the block height if that bool is set, otherwise try to fetch the receipt. 

I'm not a huge fan of the back-to-back polling loops.

Reply:
I think it would be cleaner to just separate the two loops into distinct helper functions, one that waits on the receipt, and then one that wait for a specified block height

Reply:
Split the polls into helpers 👍